backend.native/tests/interop/objc/smoke.kt

```diff
@@ -67,11 +67,12 @@ fun run() {
 
     // hashCode (directly):
     // hash() returns value of NSUInteger type.
-    val hash = when (Platform.osFamily) {
+    val hash = if (Platform.osFamily == OsFamily.WATCHOS && Platform.cpuArchitecture.bitness == 32) {
```

Comment:
This check is not correct for `watchos_arm64` (its bitness is 64 and NSInteger is defined as int). Though we are not testing this target, this piece of code might be confusing for readers.

Reply:
Changed the check